src/gd_avif.c

```diff
@@ -0,0 +1,667 @@
+/**
+ * File: AVIF IO
+ *
+ * Read and write AVIF images using libavif (https://github.com/AOMediaCodec/libavif) .
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <limits.h>
+#include <math.h>
+
+#include "gd.h"
+#include "gd_errors.h"
+#include "gdhelpers.h"
+
+#ifdef HAVE_LIBAVIF
+#include "avif.h"
+#include "internal.h"
+// internal.h is only used for AVIF_MIN() and AVIF_CLAMP() macros
+
+/*
+  Define defaults for encoding images:
+    DEFAULT_CHROMA_SUBSAMPLING: 4:2:0 is commonly used for Chroma subsampling.
+    DEFAULT_MIN_QUANTIZER, DEFAULT_MAX_QUANTIZER: 
+      We need more testing to really know what quantizer settings are optimal,
+      but teams at Google have been using maximum=30 as a starting point.
+    DEFAULT_QUALITY: following gd conventions, -1 indicates the default.
+    DEFAULT_SPEED: AVIF_SPEED_DEFAULT is -1. This simply tells the AVIF encoder to use the default speed.
+*/
+
+#define DEFAULT_CHROMA_SUBSAMPLING AVIF_PIXEL_FORMAT_YUV420
+#define DEFAULT_QUANTIZER 30
+#define DEFAULT_QUALITY -1
+#define DEFAULT_SPEED AVIF_SPEED_DEFAULT
+
+// This initial size for the gdIOCtx is standard among GD image conversion functions
```

Comment:
comments should be full sentences, so put periods at the end of them.  should double check all comments so i don't have to flag them.

Reply:
done!